src/jaegertracing/net/IPAddress.cpp

```diff
@@ -15,38 +15,92 @@
  */
 
 #include "jaegertracing/net/IPAddress.h"
+#include "jaegertracing/net/Socket.h"
 
-#include <ifaddrs.h>
 #include <sys/types.h>
 
+#ifndef WIN32
+#include <ifaddrs.h>
+#endif
+
 namespace jaegertracing {
 namespace net {
+
 namespace {
 
-struct IfAddrDeleter : public std::function<void(ifaddrs*)> {
-    void operator()(ifaddrs* ifAddr) const
+template <typename T>
+struct CDeleter : public std::function<void(T*)> {
+    void operator()(T* ifAddr) const
     {
         if (ifAddr) {
```

Comment:
I think both functions might work on a null argument. You can move `if` statement into `#else`.

Reply:
Sorry @isaachier but I didnt get your point here.

Reply:
Probably works:
```
void operator()(ifaddrs* ifAddr) const
{
#ifdef WIN32
    free(ifAddr);
#else
    ::freeifaddrs(ifAddr);
#endif
}
```

Definitely works:
```
#ifdef WIN32
    free(ifAddr);
#else
    if (ifAddr) {
        ::freeifaddrs(ifAddr);
    }
```
#endif
}